scripts/js/lib/api/releaseNotes.ts

```
@@ -257,3 +272,53 @@ async function writeReleaseNoteForVersion(
     await writeFile(versionPath, `${initialHeader}\n## ${versionsMarkdown}`);
   }
 }
+
+export function groupByMajorVersion(
+  versions: string[],
+): Record<string, string[]> {
+  const grouped = groupBy(versions, (v) => {
+    const match = v.match(/\[(\d+\.\d+)\]/);
+    if (!match || !match[1]) return "unknown";
+
+    const versionNumber = parseFloat(match[1]);
+    if (isNaN(versionNumber)) return "unknown";
+
+    return Math.floor(versionNumber).toString();
+  });
+
+  const sortedKeys = Object.keys(grouped)
+    .filter((key) => key !== "unknown")
+    .map(Number)
+    .sort((a, b) => b - a)
+    .map(String);
+
+  const sortedRecord: Record<string, string[]> = {};
+  for (const key of sortedKeys) {
+    const items = grouped[key];
+    if (Array.isArray(items)) {
+      sortedRecord[key] = items.sort((a, b) => {
+        const aMatch = a.match(/\[(\d+\.\d+)\]/);
+        const bMatch = b.match(/\[(\d+\.\d+)\]/);
+        const aVersion = aMatch ? parseFloat(aMatch[1]) : 0;
+        const bVersion = bMatch ? parseFloat(bMatch[1]) : 0;
```

I'd rather that we throw an error if either of these are undefined